autoload/lsp.vim

```diff
@@ -384,6 +384,15 @@ function! lsp#default_get_supported_capabilities(server_info) abort
     \ }
 endfunction
 
+function! lsp#getpid()
+  if !has('win32unix')
+    return getpid()
+  endif
+
+  let cmd = 'cat /proc/' . getpid() . '/winpid'
```

Comment:
This is not async and blocking. What happens if you send `null` in cygwin so it is synchronous.

null is part of spec so server should be able to handle it.

Reply:
@cossonleo Any update on this?